Pass extra cloud parameters when opening a hybrid project

4e6 opened the pull request:
### Pull Request Description

<!--
- Please describe the nature of your PR here, as well as the motivation for it.
- If it fixes an open issue, please mention that issue number here.
-->

close #13738

Changelog:
- update: `project/open` api to allow passing extra cloud parameters 
- update: gui to pass extra cloud parameters when starting hybrid projects
- update: project manager to pass extra cloud parameters to the started language server as env variables

### Important Notes

Manually testsed:

```
[TRACE] [2025-08-07T11:00:28.171] [org.enso.projectmanager.infrastructure.languageserver.ExecutorWithUnlimitedPool] Starting Language Server Process [JAVA_HOME="/home/dbushev/.local/share/enso/runtime/graalvm-ce-java24.0.1-24.2.0" ENSO_DATA_DIRECTORY="/home/dbushev/.local/share/enso" ENSO_AUXILIARY_LIBRARY_CACHES="/home/dbushev/.local/share/enso/dist/0.0.0-dev/lib" ENSO_RUNTIME_DIRECTORY="/run/user/1000/enso" ENSO_EDITION_PATH="/home/dbushev/enso/editions" ENSO_CONFIG_DIRECTORY="/home/dbushev/.config/enso" ENSO_HOME="/home/dbushev/enso" ENSO_LOG_DIRECTORY="/home/dbushev/.local/share/enso/log" ENSO_LIBRARY_PATH="/home/dbushev/enso/libraries" ENSO_CLOUD_PROJECT_DIRECTORY_PATH="enso://Users/dmitry.bushev+1871_csd2@enso.org" ENSO_CLOUD_PROJECT_ID="project-30rgh6svpzeNez8OU47ABoGmYtL" ENSO_CLOUD_PROJECT_SESSION_ID="projectsession-30xERUf8IGttPW396caV5xPCidd" "/home/dbushev/.local/share/enso/runtime/graalvm-ce-java24.0.1-24.2.0/bin/java" "-Djdk.graal.PrintGraph=Network" "--add-opens=java.base/java.nio=ALL-UNNAMED" "--add-opens=java.base/java.nio=ALL-UNNAMED" "--sun-misc-unsafe-memory-access=allow" "--enable-native-access=org.graalvm.truffle" "--module-path" "/home/dbushev/.local/share/enso/dist/0.0.0-dev/component" "-m" "org.enso.runner/org.enso.runner.Main" "--logger-connect" "//localhost:6000" "--server" "--root-id" "700ec030-61da-498f-954b-3e64a0a6e8e6" "--project-id" "d2ea1d59-0dd1-4d50-94a6-901260e8006e" "--path" "/home/dbushev/Documents/enso-projects/cloud-project-30rgh6svpzeNez8OU47ABoGmYtL/project_root" "--interface" "127.0.0.1" "--rpc-port" "51264" "--data-port" "61142" "--log-level" "TRACE" "--no-log-masking"]
```

```
ENSO_CLOUD_PROJECT_DIRECTORY_PATH="enso://Users/dmitry.bushev+1871_csd2@enso.org"
ENSO_CLOUD_PROJECT_ID="project-30rgh6svpzeNez8OU47ABoGmYtL" 
ENSO_CLOUD_PROJECT_SESSION_ID="projectsession-30xERUf8IGttPW396caV5xPCidd" 
```

<!--
- Mention important elements of the design.
- Mention any notable changes to APIs.
-->

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [ ] Unit tests have been written where possible.
- [ ] If meaningful changes were made to logic or tests affecting Enso Cloud integration in the libraries, 
      or the Snowflake database integration, a run of the [Extra Tests](https://github.com/enso-org/enso/actions/workflows/extra-nightly-tests.yml) has been scheduled.
  - If applicable, it is suggested to paste a link to a successful run of the Extra Tests.

4e6 committed:
update: project/open api

4e6 committed:
update: project manager api

4e6 committed:
misc: format

PabloBuchu commented on app/common/src/services/Backend.ts:
```diff
@@ -1280,13 +1280,25 @@ export interface UpdateProjectRequestBody {
+  /** Cloud project directory path. */
+  readonly cloudProjectDirectoryPath: string
+  /** Cloud project id. */
+  readonly cloudProjectId: string
```
there is `ProjectId` type

PabloBuchu commented on app/common/src/services/Backend.ts:
```diff
@@ -1280,13 +1280,25 @@ export interface UpdateProjectRequestBody {
+  /** Cloud project id. */
+  readonly cloudProjectId: string
+  /** Cloud project session id. */
+  readonly cloudProjectSessionId: string
```
there is `ProjectSessionId` type

PabloBuchu commented on app/common/src/services/Backend.ts:
```diff
@@ -1280,13 +1280,25 @@ export interface UpdateProjectRequestBody {
+ */
+export interface OpenHybridProjectParameters {
+  /** Cloud project directory path. */
+  readonly cloudProjectDirectoryPath: string
```
not sure where this comes from but we also have `EnsoPath` type

PabloBuchu commented on docs/language-server/protocol-project-manager.md:
```diff
@@ -411,7 +411,16 @@ the action.
+interface CloudParams {
+  cloudProjectDirectoryPath: string;
+  cloudProjectId: string;
+  cloudProjectSessionId: string;
```
types

4e6 committed:
update: add types

hubertp reviewed (commented):
I'm a bit undecided how cloud project id and project session id should be passed around. The launcher takes an extra parameter `project-id` for local project id and maybe it would be worth having `--cloud-project-id` and `--cloud-project-session-id` rather (or apart from?) env variables being set.

4e6 commented:
We already pass the cloud project directory through the env variable. I thought it would be reasonable to pass other parameters the same way. 
And the `project-id` is required to run the project, while the cloud parameters are only passed in case of hybrid execution

4e6 committed:
fix: interruption tests

hubertp committed:
Pass `projectId` and `projectSessionId` in MDC

Hybrid projects require cloud's `project-id` and `project-session-id` to
properly associate the project with its logs.
GUI now provides the IDs when opening hybrid projects.

This change adds IDs to MDC, so that every log message contains
necessary metadata. Unfortunately MDC does not play well with
ThreadPools, meaning that any MDC is quickly dropped after initial
Language Server startup.
This change does not attempt to modify every thread creation to pass
around MDCs. Instead this change stores MDC with necessary IDs during
central socket server setup and appends it to every log message that
needs to be forwarded.
As logged `ILoggingEvent` does not allow to append MDC per event, a
proxy was defined.

hubertp commented:
With https://github.com/enso-org/enso/pull/13744/commits/00ba885a454bddab04ce2cc685fa822acdde671f in place, hybrid projects will now have extra `metadata` entries in logs:
- `metadata.projectId`
- `metadata.projectLocalId`
- `metadata.projectSessionId`

hubertp committed:
fix tests

hubertp reviewed (approved):
Approving the changes in backend that I haven't committed myself ;p
